components/authn-service/server/server.go

```diff
@@ -177,6 +166,35 @@ func newServer(ctx context.Context, c Config) (*Server, error) {
 	// make grpc-go log through zap
 	grpclog.SetLoggerV2(wrap.WrapZapGRPC(s.logger))
 
+	// Add the legacy data collector token as a secret if it was defined in the config.
+	if c.LegacyDataCollectorToken != "" {
+		var tokenID string
+		if existingID, err := ts.GetTokenIDWithValue(ctx, c.LegacyDataCollectorToken); err != nil {
+			// If we couldn't find the legacy data collector token, create it.
+			if _, ok := errors.Cause(err).(*tokens.NotFoundError); ok {
+				var token *tokens.Token
+				token, err = ts.CreateLegacyTokenWithValue(ctx, c.LegacyDataCollectorToken)
+				tokenID = token.ID
+			}
+
+			if err != nil {
+				return nil, errors.Wrap(err,
+					"could not populate the legacy data collector token")
+			}
+		} else {
+			tokenID = existingID
+		}
+
+		_, err := policiesClient.AddPolicyMembers(ctx, &authz.AddPolicyMembersReq{
+			Id:      IngestAccessPolicyID,
+			Members: []string{fmt.Sprintf("token:%s", tokenID)},
+		})
+		if err != nil {
+			s.logger.Warn(errors.Wrap(err, "there was an error granting the legacy data collector token ingest access").Error())
+			s.logger.Warn(fmt.Sprintf("please manually add token with ID %s to the policy with ID %s", tokenID, IngestAccessPolicyID))
```

Comment:
Will be slightly clearer:
```suggestion
			s.logger.Warn(fmt.Sprintf("please manually add token with ID %q to the policy with ID %q", tokenID, IngestAccessPolicyID))
```

Reply:
Don't we have a Warnf? (or is this the zap logger...?) :thinking:

Reply:
Zap :wee: